capstone/src/main/webapp/script.js

```diff
@@ -31,14 +32,150 @@ async function submitInput() {
   if (!langElement) {
     return;
   }
+
+  // Get the selected attributes
   const attributes = document.getElementById("available-attributes").getElementsByTagName('input');
-  const requestedAttributes = []
+  const requestedAttributes = [];
   for (const attribute of attributes) {
     if (attribute.checked) {
       requestedAttributes.push(attribute.value);	
     }	
   }
-  await callPerspective(textElement.value, langElement.value, requestedAttributes);
+
+  // Get the selected analysis type
+  document.getElementById('analysis-container').innerHTML = '';
+  const radios = document.getElementsByName('analysisRadios');
+  var delimiter = "";
+  for (i = 0; i < radios.length; i++) {
+    if (radios[i].checked) {
+      delimiter = radios[i].value;
+      break;
+    }
+  }
+
+  handleInput(textElement.value, langElement.value, requestedAttributes, delimiter);
+}
+
+/** Submits the input to Perspective and loads the appropriate output */
+async function handleInput(text, lang, requestedAttributes, delimiter) {
+  // Draw the separating line for the output
+  const separator = document.getElementById('separator-container');
+  separator.innerHTML = '';
+  separator.appendChild(document.createElement('hr')); 
+	
+  // Make Perspective call for the entire submission and load graph data
+  const toxicityData = await callPerspective(text, lang, requestedAttributes);
+  loadChartsApi(toxicityData);
+
+  // Get detailed analysis if requested
+  if (delimiter != "") {
+    getAnalysis(text, lang, requestedAttributes, delimiter); 
+  }
+}
+
+/** Prints detailed analysis of text by word or sentence */
+async function getAnalysis(text, lang, requestedAttributes, delimiter) {
```

Comment:
nit: This function is a bit long. Consider breaking up some of the functionality into separate functions (e.g. the part where you set the color of the substring).

Reply:
Fixed - I added 2 more helper functions `colorSubstring` and `sortAttributes`